### PR TITLE
Li/add docs api

### PR DESCRIPTION
**Pull request by wanliAlex**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
we move the query parameters `non_tensor_fields`, `use_existing_tensors`, `image_download_headers`, `model_auth`, `mappings` to the request body.

* **What is the current behavior?** (You can also link to an open issue here)
`non_tensor_fields`, `use_existing_tensors`, `image_download_headers`, `model_auth`, `mappings` are query parameters

* **What is the new behavior (if this is a feature change)?**
they are now moved to the query body as `nonTensorFields, useExistingTensors, imageDownloadHeaders, modelAuth, mappings`

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
yes. 
for py-marqo users, they need to upgrade the py-marqo version
for bash curl users, they need to change their requests

marqo will still support old APIs recently but my remove them in the near future.

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)
running

* **Related Python client changes** (link commit/PR here)
working on

* **Related documentation changes** (link commit/PR here)
finished

* **Other information**:

* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: src/marqo/client.py [hunk 1]
```diff
@@ -9,6 +9,11 @@
 from marqo._httprequests import HttpRequests
 from marqo import utils, enums
 from marqo import errors
+from marqo.version import __minimum_supported_marqo_version__
```
Comment (farshidz) [inline src/marqo/client.py]: Let's add a method `marqo.version.minimum_supported_marqo_version()` to access this value. Direct access to private fields is bad practice

File: src/marqo/client.py [hunk 2]
```diff
@@ -189,4 +196,13 @@ def get_cuda_info(self):
...
 
+    def _marqo_minimum_supported_version_check(self):
+        if self.url not in marqo_url_and_version_cache:
+            marqo_url_and_version_cache[self.url] = self.get_marqo()["version"]
+        marqo_version = marqo_url_and_version_cache[self.url]
+        if marqo_version < __minimum_supported_marqo_version__:
+            mq_logger.warning(f"Your Marqo python client requires a minimum Marqo version of "
+                              f"{__minimum_supported_marqo_version__} to function properly, while your Marqo version is {marqo_version}. "
+                              f"Please upgrade your Marqo instance to avoid potential errors. "
+                              f"If you have already upgraded your Marqo instance but still having this warning, please import you Marqo python client again and retry.")
```
Comment (farshidz) [inline src/marqo/client.py]: If you have already upgraded your Marqo instance, please import your Marqo Python client again and retry.

Comment (farshidz) [inline src/marqo/client.py]: Python must be capitalised 

Your Marqo Python client requires a minimum Marqo version of

File: src/marqo/index.py [hunk 3]
```diff
@@ -460,7 +459,8 @@ def verbosely_add_docs(i, docs):
             errors_detected = False
 
             t0 = timer()
-            res = self.http.post(path=path_with_query_str, body=docs, index_name=self.index_name,)
+            body = {"documents": docs, **base_body}
+            res = self.http.post(path=path_with_query_str, body=body)
```
Comment (farshidz) [inline src/marqo/index.py]: add `index_name` so that Cloud v2 requests go directly to index

File: src/marqo/index.py [hunk 4]
```diff
@@ -515,6 +515,10 @@ def verbosely_add_docs(i, docs):
         mq_logger.debug('completed batch ingestion.')
         return results
 
+    def get_settings(self) -> dict:
```
Comment (farshidz) [inline src/marqo/index.py]: Duplicate method. Looks like a merge conflict resolved incorrectly? Please keep the mainline version that passed `index_name`

File: tests/v0_tests/test_add_documents.py [hunk 5]
```diff
@@ -504,14 +515,45 @@ def test_multimodal_combination_doc(self):
...
+        except MarqoApiError as s:
+            pass
+
+    def tearDown(self) -> None:
+        try:
+            self.client.delete_index(self.index_name_1)
+        except MarqoApiError as s:
+            pass
+
+    def test_add_docs_image_download_headers(self):
```
Comment (farshidz) [inline tests/v0_tests/test_add_documents.py:530]: I think this test can go under `TestAddDocuments` and doesn't require a new class

File: tests/v0_tests/test_client.py [hunk 6]
```diff
@@ -33,3 +29,64 @@ def test_health(self):
...
+        self.assertIn(self.client_settings['url'], marqo_url_and_version_cache)
+        assert marqo_url_and_version_cache[self.client_settings['url']] == '0.0.0'
+
+    def test_version_check_multiple_instantiation(self):
+        """Ensure that duplicated instantiation of the client does not result in multiple APIs calls of get_marqo()"""
+        with mock.patch("marqo.client.Client.get_marqo") as mock_get_marqo:
+            mock_get_marqo.return_value = {'version': '0.0.0'}
+            client = Client(**self.client_settings)
+
+        mock_get_marqo.assert_called_once()
```
Comment (farshidz) [inline tests/v0_tests/test_client.py:63]: Indentation doesn't look right. These should be within the context manager and the context manager should release the mock without requiring an explicit release

File: tests/v0_tests/test_client.py [hunk 7]
```diff
@@ -33,3 +29,64 @@ def test_health(self):
...
+        mock_get_marqo.reset_mock()
+
+        for _ in range(10):
+            with mock.patch("marqo.client.mq_logger.warning") as mock_warning, \
+                 mock.patch("marqo.client.Client.get_marqo") as mock_get_marqo:
+
+                client = Client(**self.client_settings)
+
+            mock_get_marqo.assert_not_called()
```
Comment (farshidz) [inline tests/v0_tests/test_client.py]: Indentation doesn't look right. These should be within the context manager and the context manager should release the mock without requiring an explicit release

File: tests/v0_tests/test_client.py [hunk 8]
```diff
@@ -33,3 +29,64 @@ def test_health(self):
...
+        warning_message = mock_warning.call_args[0][0]
+
+        # Assert the message is what you expect
+        self.assertIn("Please upgrade your Marqo instance to avoid potential errors.",  warning_message)
+
+        # Assert the url is in the cache
+        self.assertIn(self.client_settings['url'], marqo_url_and_version_cache)
+        assert marqo_url_and_version_cache[self.client_settings['url']] == '0.0.0'
+
+    def test_version_check_multiple_instantiation(self):
```
Comment (farshidz) [inline tests/v0_tests/test_client.py:57]: These can be under `TestClient` without a new class

**Review by farshidz — CHANGES_REQUESTED** [review]

File: src/marqo/client.py (hunk 1 above)
Comment (wanliAlex) [reply src/marqo/client.py]: thanks! resolved

File: src/marqo/index.py (hunk 3 above)
Comment (wanliAlex) [reply src/marqo/index.py]: thanks!

**Review by farshidz — APPROVED** [review]

**Pull request merged**